src/include/sof/schedule/ll_schedule_domain.h

```diff
@@ -106,51 +125,65 @@ static inline int domain_register(struct ll_schedule_domain *domain,
 
 	ret = domain->ops->domain_register(domain, period, task, handler, arg);
 
+	if (!ret) {
```

Comment:
I dont understand why we are increasing ref counts if registration fails ?

Reply:
> I dont understand why we are increasing ref counts if registration fails ?

ret == 0 means succeed.